scripts/src/javatests/org/oppia/android/scripts/docs/KdocValidityCheckTest.kt

```diff
@@ -0,0 +1,875 @@
+package org.oppia.android.scripts.docs
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.oppia.android.scripts.proto.KdocValidityExemptions
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+
+/** Tests for [KdocValidityCheck]. */
+class KdocValidityCheckTest {
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+  private val KDOC_CHECK_PASSED_OUTPUT_INDICATOR = "KDOC VALIDITY CHECK PASSED"
+  private val KDOC_CHECK_FAILED_OUTPUT_INDICATOR = "KDOC VALIDITY CHECK FAILED"
+  private val pathToProtoBinary = "scripts/assets/kdoc_validity_exemptions.pb"
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("testfiles")
+    tempFolder.newFolder("scripts", "assets")
+    tempFolder.newFile(pathToProtoBinary)
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testKdoc_function_withKdoc_checkShouldPass() {
+    val testContent =
+      """
+      /** 
+       * Returns the string corresponding to this error's string resources, or null if there 
+       * is none.
+       */
+      fun getErrorMessageFromStringRes(context: Context): String? {
+        return error?.let(context::getString)
+      }
+      """.trimIndent()
+    val tempFile = tempFolder.newFile("testfiles/TempFile.kt")
+    tempFile.writeText(testContent)
```

Comment:
you can create a helper function at bottom of this file and use it here for writing text file

Reply:
Hmm, we just want to use `writeText` here in order to write the test file, so do we actually need to create a separate function for that? Also, this will make this script a bit inconsistent with all the other scripts, as there we are following the current approach.